src/common/column_matrix.h

```diff
@@ -366,19 +423,107 @@ class ColumnMatrix {
       using ColumnBinT = decltype(t);
       ColumnBinT* local_index = reinterpret_cast<ColumnBinT*>(index_.data());
       size_t const batch_size = batch.Size();
-      size_t k{0};
-      for (size_t rid = 0; rid < batch_size; ++rid) {
-        auto line = batch.GetLine(rid);
-        for (size_t i = 0; i < line.Size(); ++i) {
-          auto coo = line.GetElement(i);
-          if (is_valid(coo)) {
-            auto fid = coo.column_idx;
-            const uint32_t bin_id = row_index[k];
-            SetBinSparse(bin_id, rid + base_rowid, fid, local_index);
-            ++k;
+
+      // Parallel sparse batch processing
+      dmlc::OMPException exc;
+      std::vector<size_t> n_elements((n_threads + 1) * n_features, 0);
+      std::vector<size_t> k_offsets(n_threads + 1, 0);
+      size_t block_size = DivRoundUp(batch_size, n_threads);
+
+      /*
+       * We use bitfield as a missing indicator. To ensure thread safe access to the bitfield
+       * each underlying word of the bitfiled should be processed by a single thread.
+       * So we need to align the row-blocks.
+       */
+      block_size = DivRoundUp(block_size, MissingIndicator::BitFieldT::kValueSize) *
+                   MissingIndicator::BitFieldT::kValueSize;
```

Comment:
Improving the readability a bit:
```suggestion
      const auto word32size = MissingIndicator::BitFieldT::kValueSize;

      size_t block_size = DivRoundUp(batch_size, n_threads);   // preliminary block size
      // align block_size to be multiple of kValueSize, so that each thread processes full words in the bitfield
      block_size = DivRoundUp(block_size, word32size) * word32size;
```